Neos.ContentRepository/src/Feature/NodeReferencing/NodeReferencing.php

```diff
@@ -83,34 +132,30 @@ public function handleSetNodeReferences(SetNodeReferences $command): CommandResu
         $this->getNodeAggregateEventPublisher()->withCommand(
             $command,
             function () use ($command, &$domainEvents, $sourceNodeAggregate) {
-                $events = [];
                 $sourceNodeType = $this->requireNodeType($sourceNodeAggregate->getNodeTypeName());
-                $declaration = $sourceNodeType->getProperties()[$command->referenceName->value]['scope'] ?? null;
-                if (is_string($declaration)) {
-                    $scope = PropertyScope::from($declaration);
-                } else {
-                    $scope = PropertyScope::SCOPE_NODE;
-                }
+                $scopeDeclaration = $sourceNodeType->getProperties()[(string)$command->referenceName]['scope'] ?? '';
+                $scope = PropertyScope::tryFrom($scopeDeclaration) ?: PropertyScope::SCOPE_NODE;
+
                 $affectedOrigins = $scope->resolveAffectedOrigins(
                     $command->sourceOriginDimensionSpacePoint,
                     $sourceNodeAggregate,
                     $this->interDimensionalVariationGraph
                 );
-                foreach ($affectedOrigins as $originDimensionSpacePoint) {
-                    $events[] = DecoratedEvent::addIdentifier(
-                        new NodeReferencesWereSet(
-                            $command->contentStreamIdentifier,
-                            $command->sourceNodeAggregateIdentifier,
-                            $originDimensionSpacePoint,
-                            $command->destinationNodeAggregateIdentifiers,
-                            $command->referenceName,
-                            $command->initiatingUserIdentifier
-                        ),
-                        Uuid::uuid4()->toString()
-                    );
-                }
-
-                $domainEvents = DomainEvents::fromArray($events);
+
+                $event = DecoratedEvent::addIdentifier(
+                    new NodeReferencesWereSet(
+                        $command->contentStreamIdentifier,
+                        $command->sourceNodeAggregateIdentifier,
+                        $affectedOrigins,
```

Comment:
ah, now I understand it. Still I guess we should add a comment in the event what the semantics of multiple affectedOrigins are.